refbox/src/tournament_manager/mod.rs

```diff
@@ -1715,6 +1775,174 @@ impl TournamentManager {
         }
     }
 
+    pub fn pause_for_confirm(&mut self, now: Instant) -> Result<()> {
+        if self.timeout_state.is_some() {
+            return Err(TournamentManagerError::PausingDuringTimeout);
+        }
+        if !self.clock_state.is_running() {
+            return Err(TournamentManagerError::ClockStopped);
+        }
+        info!("Pausing for Confirmation");
+        let pause_inst = match self.clock_state {
+            ClockState::CountingDown {
+                start_time,
+                time_remaining_at_start,
+            } => min(start_time + time_remaining_at_start, now),
+            ClockState::CountingUp { .. } => now,
+            ClockState::Stopped { .. } => unreachable!(),
+        };
+
+        let dur_pause = match self.current_period {
+            GamePeriod::SecondHalf => {
+                if self.config.overtime_allowed {
+                    let mut len = min(self.config.pre_overtime_break, self.config.minimum_break);
+                    len /= 2;
+                    len
+                } else if self.config.sudden_death_allowed {
+                    let mut len = min(
+                        self.config.pre_sudden_death_duration,
+                        self.config.minimum_break,
+                    );
+                    len /= 2;
+                    len
+                } else {
+                    self.config.minimum_break / 2
+                }
+            }
+            GamePeriod::OvertimeSecondHalf => {
+                if self.config.sudden_death_allowed {
+                    let mut len = min(
+                        self.config.pre_sudden_death_duration,
+                        self.config.minimum_break,
+                    );
+                    len /= 2;
+                    len
+                } else {
+                    self.config.minimum_break / 2
+                }
+            }
+            GamePeriod::SuddenDeath => self.config.minimum_break / 2,
+            _ => {
+                unreachable!()
+            }
+        };
+
+        info!("Current Period: {}", self.current_period);
+
+        let clock_at_pause = match self.current_period {
+            GamePeriod::SuddenDeath => self.clock_state.clock_time(now).unwrap(),
+            GamePeriod::SecondHalf | GamePeriod::OvertimeSecondHalf => Duration::from_millis(10),
+            _ => unreachable!(),
```

Comment:
I would prefer listing out each of the other possible values of current period here rather than using `_`. This makes it easier to catch issues if the GamePeriod enum gets modified in the future